learn/appchains/tx-fees.md

```diff
@@ -0,0 +1,116 @@
+
+### Weight {: #baseline-weight}
+
+Broadly speaking, weight refers to the execution time it takes to validate a block, measured in picoseconds. Weight is divided into two separate variables:
+
+- **refTime** - corresponds to the weight associated with computation time and database reads/writes
+- **proofSize** - corresponds to the weigth associated to the size of the _Proof-Of-Validity_ (or _PoV_ for short). The _PoV_ is associated with the relevant state of a transaction, and it is what the Appchain block producer shares with the relay chain validator to get its block finalized as part of the [Appchain transaction flow](/learn/appchains/overview/#appchain-transaction){target=\_blank}
+
+To find the weights for all function calls, they are benchmarked in a system with reference hardware, and the approximate values of _refTime_ and _proofSize_ are set. This process is repeated for all function calls that consume blockspace and affect the _PoV_.
+
+For transactions in which the fees are handled by [Transaction Payment](https://docs.rs/pallet-transaction-payment/latest/pallet_transaction_payment/){target=\_blank} module, all weight-based parameters are then passed through a _WeightToFee_ algorithm that converts all to a final value, deducted from the sender's account when executing the function call. The algorithm can be customized, but Tanssi Appchains have a constant value set.
+
+For EVM transactions, gas is converted to weight through a _GastoWeight_ algorithm so that all EVM calls can be mapped to block execution time. Nevertheless, fees are handled at an EVM execution level.
+
+### Baseline Transaction Fees {: #baseline-transaction-fees}
+
+<!-- https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/transaction-payment/src/lib.rs#L627-L652 -->
+
+With all function calls benchmarked, the transaction fee for each specific call can be obtained. Transaction fees are typically comprised of the following elements:
+
+- **Base Fee** - baseline cost for a transaction to be included. It accounts for the transaction inclusion overhead, like signature verification. Two separate parameters define it:
+    - **ExtrinsicBaseWeight** - a constant value that represents the weight of the transaction inclusion overhead
+    - **WeightToFee** - a polynomial function that converts weight to fee
+- **Weight Fee** - fee defined by two separate parameters:
+    - **Benchmarked Weight** - weight that accounts for the complexity (execution time) of a specific call
+    - **Congestion Multiplier** - a function that converts weight to fee and can be adjusted to account for the congestion of the network (weight consumed in the previous block). For Tanssi Appchains, this is a constant
+- **Length Fee** - a function that defines the per-byte fee (_LenghtToFee_) correlated to the length in bytes of the function call. It can range from a constant to a cubic curve. For Tanssi Appchains, this is a constant
```

Comment:
```suggestion
- **Length Fee** - a function that defines the per-byte fee (_LengthToFee_) correlated to the length in bytes of the function call. It can range from a constant to a cubic curve. For Tanssi Appchains, this is a constant
```